src/couch_replicator_tasks.erl

```diff
@@ -0,0 +1,85 @@
+-module(couch_replicator_tasks).
+
+-behavior(gen_server).
+
+-export([all/0]).
+-export([start_link/0]).
+-export([init/1, terminate/2, code_change/3]).
+-export([handle_call/3, handle_cast/2, handle_info/2]).
+
+-include_lib("couch/include/couch_db.hrl").
+-include("couch_replicator.hrl").
+-include("couch_replicator_api_wrap.hrl").
+
+
+all() ->
+    gen_server:call(?MODULE, all).
+
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+
+init([]) ->
+    {ok, ok}.
+
+
+handle_call(all, _From, State) ->
+    Jobs = ets:tab2list(couch_replicator_scheduler),
+    JobProps = lists:map(fun(Job) ->
+        Rep = Job#job.rep,
+        Source = case Rep#rep.source of
+            #httpdb{} -> (Rep#rep.source)#httpdb.url;
+            Source0 -> Source0
+        end,
+        Target = case Rep#rep.target of
+            #httpdb{} -> (Rep#rep.target)#httpdb.url;
+            Target0 -> Target0
+        end,
+        History = lists:map(fun(Event) ->
+            EventProps  = case Event of
+                {{crashed, Reason}, _When} ->
+                    [{type, crashed}, {reason, Reason}];
+                {Type, _When} ->
+                    [{type, Type}]
+            end,
+            {_Type, {Mega, Seconds, Micro}} = Event,
+            UnixTime = Mega*1000000+Seconds+Micro/1000000,
+            {[{timestamp, UnixTime} | EventProps]}
+        end, Job#job.history),
+        {BaseID, Ext} = Job#job.id,
+        User = case (Rep#rep.user_ctx)#user_ctx.name of
+            null ->
+                case cloudant_util:customer_name(Rep#rep.db_name) of
```

Comment:
good catch. customer_name is part of multi-tenancy, which is not part of couchdb.